flocker/control/_model.py

```diff
@@ -251,10 +251,23 @@ def applications(self):
             for application in node.applications:
                 yield application
 
+    def update_node(self, node):
+        """
+        Replace existing ``Node`` with updated version.
```

Comment:
Perhaps the docstring could be a little clearer that this returns an updated copy of this Deployment rather than updating in-place.